File: Python/S3/S3-1.py
import scipy.stats as stats

# Datos del problema
mu0 = 880000   # Hipótesis nula
xbar = 850000  # Media muestral
sigma = 32000  # Desviación estándar poblacional
n = 25         # Tamaño de la muestra
alpha = 0.01   # Nivel de significancia (1 - nivel de confianza)

# Cálculo del estadístico de prueba
z = (xbar - mu0) / (sigma / (n**0.5))

# Valor crítico
z_critico = stats.norm.ppf(1 - alpha/2)

# Comparación
if abs(z) > z_critico:
    print("Rechazar H0")
else:
    print("No rechazar H0")

# Opcional: Mostrar valores calculados
print(f"\nEstadístico Z calculado: {z:.4f}")
print(f"Valor crítico Z: ±{z_critico:.4f}")

File: R/S3-1.R
# Script de R para realizar un z-test completo

# Datos de la muestra y población
mu <- 880          # media poblacional
sample_mean <- 850 # media muestral
sigma <- 32        # desviación estándar poblacional
n <- 25            # tamaño de la muestra

# Instalar y cargar la librería necesaria
if (!require("BSDA")) install.packages("BSDA")
library(BSDA)

# Realizar el z-test
z_test_result <- z.test(x = rep(sample_mean, n), mu = mu, sigma.x = sigma)

# Estadístico crítico para un intervalo de confianza del 99%
alpha <- 0.01
z_critical <- qnorm(alpha / 2, lower.tail = FALSE)  # Valor absoluto

# Extraer el estadístico de prueba
z_stat <- z_test_result$statistic

# Comparación del estadístico de prueba con el estadístico crítico
if (abs(z_stat) > z_critical) {
  cat(sprintf("Rechazamos la hipótesis nula. Estadístico de prueba: %.2f, Estadístico crítico: ±%.2f\n", z_stat, z_critical))
} else {
  cat(sprintf("No rechazamos la hipótesis nula. Estadístico de prueba: %.2f, Estadístico crítico: ±%.2f\n", z_stat, z_critical))
}

# Mostrar resultados completos
print(z_test_result)
cat("\nValor crítico (Z) para 99% de confianza:", z_critical, "y", -z_critical)
